### ISSUE TITLE
Bug: `Boxes.merge` keeps stale list padding, so `to_tensor()` drops the merged boxes

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4175]
## Summary

`Boxes.merge` builds its result with `obj = self.clone()` (`kornia/geometry/boxes.py:342` at `8d685a37`) and then overwrites `obj._data` with the concatenation. `clone()` copies `_N` — the per-element padding counts recorded when the object was built from a list — but the merge just changed how many boxes each row holds, so `_N` is stale afterwards.

`to_tensor` trims with those counts:

```python
boxes = [torch.nn.functional.pad(o, ... + [0, -n]) for o, n in zip(boxes, self._N)]
```

so a list-backed `Boxes` silently **drops the boxes that were just merged in**, and returns padding entries in their place. Nothing raises.

## Reproduction

```python
import torch
from kornia.geometry.boxes import Boxes

first = torch.tensor([[[1.0, 2.0], [4.0, 2.0], [4.0, 3.0], [1.0, 3.0]]])   # 1 box
second = torch.cat([first, first])                                          # 2 boxes
lb = Boxes([first, second])          # _N == [1, 0], data (2, 2, 4, 2)

merged = lb.merge(Boxes(torch.zeros(2, 1, 4, 2) + 7.0))
print(merged._N, tuple(merged.data.shape))
for i, row in enumerate(merged.to_tensor("xyxy")):
    print(i, row.tolist())
print(merged.get_boxes_shape()[0].tolist())
```

## Observed

```
[1, 0] (2, 3, 4, 2)
0 [[1.0, 2.0, 5.0, 4.0], [0.0, 0.0, 1.0, 1.0]]
1 [[1.0, 2.0, 5.0, 4.0], [1.0, 2.0, 5.0, 4.0], [7.0, 7.0, 8.0, 8.0]]
[[2.0, 1.0, 1.0], [2.0, 2.0, 1.0]]
```

Row 0 should be its original box plus the merged `[7., 7., 8., 8.]`. Instead the merged box is trimmed off and the phantom padding box `[0., 0., 1., 1.]` is returned in its place — the data is there (`to_tensor(..., as_padded_sequence=True)` shows all three), only the bookkeeping says to cut it. Row 1 had `_N == 0` and is correct, so the corruption is per-element and depends on how ragged the original list was.

## Expected

`merge` updates `_N` by the number of boxes added — `[n + boxes.data.shape[-3] for n in self._N]` for a batched incoming object, i.e. the pad counts stay relative to the new row lengths — or, if merging into a list-backed object is not meant to be supported, it raises instead of returning a silently trimmed object. `index_put` takes the same `self.clone()` path but does not change the box count, so it is unaffected.

## Context

Found while reviewing #4060 (bbox/boxes convention blocks, batch 4a). That PR documents `merge` and adds a convention pin for it, but the pin exercises the batched tensor path only, and the batch is documentation and tests only, so this is out of scope there. Distinct from #4168, which is `merge`'s unconditional `dim=1` on unbatched input — same method, different defect; both are fixable in one PR. Part of the #4020 audit.

## Environment

kornia 0.9.0rc1 @ `8d685a37` (main), torch 2.9.1, python 3.11, cpu.

Posted on behalf of @ducha-aiki by Claude (Opus 5).

**Comment by cqxswbc2** [comment]
I’d like to take this. I’ll preserve the list-backed semantics by repacking each batch row as existing real boxes, incoming boxes, then trailing padding, instead of clearing _N and exposing the old pad rows as real boxes. I’ll cover both return and in-place paths with a regression test that checks values and return type.
